pkg/azureutils/azure_disk_utils.go

```diff
@@ -1279,6 +1246,77 @@ func UpdateCRIWithRetry(ctx context.Context, informerFactory azdiskinformers.Sha
 	return updatedObj, err
 }
 
+func getOriginalObjFromClientForUpdate(ctx context.Context, w workflow.Workflow, informerFactory azdiskinformers.SharedInformerFactory, cachedClient client.Client, azDiskClient azdisk.Interface, obj client.Object) (client.Object, error) {
+	var err error
+	var originalObj client.Object
+	objName := obj.GetName()
+
+	w.Logger().V(5).Infof("getting the original object (%s) from clients", objName)
+
+	switch target := obj.(type) {
+	case *azdiskv1beta2.AzVolume:
+		if azDiskClient == nil {
+			return nil, status.Errorf(codes.Internal, "cannot update AzVolume object if azDiskClient is not provided.")
```

Comment:
This check and error should be performed by `UpdateCRIWithRetry` since that is what would perform the update. It is not required for this method to function if `informerFactory` or `cachedClient` is provided.

Reply:
Yes, it would make more sense to check clients inside `UpdateCRIWithRetry`. Let me move it to line 1133, same place I checked for originalObj.